src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -93,13 +97,48 @@ export const MultiSelect: ChakraComponent<
         ...rest
       } = props;
 
+      const [userClickedOutside, setUserClickedOutside] =
+        useState<boolean>(false);
+
       // Create a ref to hold a reference to the accordian button, enabling us
       // to programmatically focus it.
       const accordionButtonRef: React.RefObject<HTMLDivElement> =
         useRef<HTMLDivElement>();
+      const containerRef: React.RefObject<HTMLDivElement> =
+        useRef<HTMLDivElement>();
       const expandToggleButtonRef: React.RefObject<HTMLButtonElement> =
         useRef<HTMLButtonElement>();
 
+      const handleEvent = (e) => {
+        if (
+          e.type === "mousedown" ||
+          (e.type === "keydown" && e.code === "Enter")
+        ) {
+          if (
+            containerRef.current &&
+            !containerRef.current.contains(e.target)
+          ) {
+            setUserClickedOutside(true);
+          } else {
+            setUserClickedOutside(false);
+          }
+        } else {
+          setUserClickedOutside(false);
```

Comment:
Do you need to fire these events for any and all key presses? I think you can get away with removing line 126 but you do need it on line 123.